### ISSUE TITLE
HEPMC3 missing in CMakeConfig file

### ISSUE THREAD

**Issue by guiguem** [WCSim/WCSim#461]
With HEPMC3 being a required dependency, the cmake config needs to be updated to reflect this, so that the downstream packages that uses WCSim don't need to have included explicitly (unless they need it).

https://github.com/WCSim/WCSim/blob/5bf9a0bba53960643b49cbfe625e6c32b05e0874/cmake/WCSimRootConfig.cmake.in#L6C1-L6C16

We might need to be slightly clever about this, as for WCSimRoot build HEPMC3 is not needed.

**Comment by tdealtry** [comment]
Agreed that this would be very good!

On the other hand, it might just be WCSimFQTuner that needs to be a direct WCSim dependency (the others like fiTQun, TriggerApp, BONSAI, FLOWER, LEAF, etc. just need WCSimRoot). So if this turns out to be complicated to implement, it's probably not worth spending a lot of time trying

**Comment by guiguem** [comment]
True, a possibility (a lot simpler) would be to have two config.cmake template (one for WCSim and one for WCSimRoot).
I will give it a try in the coming days then.

**Comment by tdealtry** [comment]
Thanks @guiguem !
